src/tests/Regressions/coreclr/GitHub_110987/test110987.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Runtime.Loader;
+using Xunit;
+
+class TestAssemblyLoadContext : AssemblyLoadContext
+{
+    public TestAssemblyLoadContext() : base(isCollectible: true)
+    {
+    }
+}
+
+public class Test110987
+{
+    [Fact]
+    public static void TestDynamicMethodALC()
+    {
+        string tempFilePath = Path.GetTempFileName() + ".dll";
```

Comment:
If we need to save files to disk, we should make sure to clean up after the test.

Instead of saving the assembly to disk, you can save it to MemoryStream and load it from MemoryStream. It would avoid the problems with left-over temporary files and make the test more reliable (avoid interactions with antiviruses, etc.).